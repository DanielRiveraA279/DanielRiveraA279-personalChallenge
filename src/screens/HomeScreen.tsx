import React, { FC, useEffect, useState } from 'react';
import { StyleSheet, Text, View, ActivityIndicator, SafeAreaView, ScrollView, FlatList, Alert, RefreshControl, Image, TextInput, Button, NativeSyntheticEvent, TextInputChangeEventData } from 'react-native';
import Moment from 'moment';
import { useWeather } from "../hooks/useWeather";
import { Usedebouncevalue } from "../hooks/useDebouncedValue";
import { styles } from '../styles/general';
import { InputSearch } from '../components/InputSearch';
import { GradientBackground } from '../components/GradientBackground';
import { PickerSearch } from '../components/PickerSearch';

export const HomeScreen = () => {
    const { weatherList, getWeather } = useWeather();
    const [textValue, setTextValue] = useState('');
    const debouncedValue = Usedebouncevalue(textValue, 500);

    useEffect(() => {
       getWeather(debouncedValue);
    }, [debouncedValue]);

    return (
        <GradientBackground>
            <ScrollView>
                {Object.values(weatherList).length >= 1 ?
                    <Text style={styles.title}>CLIMA</Text> :
                    <Text style={styles.subtitle}>BUSCAR CLIMA</Text>
                }
                {Object.values(weatherList).length >= 1 && <View style={styles.current}>
                    <Image
                        style={styles.largeIcon}
                        source={{
                            uri: `http://openweathermap.org/img/wn/${weatherList[0]?.weather[0]?.icon}@4x.png`,
                        }}
                    />
                    <Text style={styles.textDetail}>{Math.round(weatherList[0]?.main.temp)}°C</Text>
                </View>}

                {Object.values(weatherList).length >= 1 && <Text style={styles.subtitle}>{weatherList[0]?.weather[0]?.description.toUpperCase()}</Text>}

                <View>
                    <PickerSearch
                        select={debouncedValue}
                        onValueChange={setTextValue}
                    />
                    {/* <InputSearch
                        placeholder="Ciudad Actual Buenos Aires"
                        placeholderTextColor="white"
                        value={textValue}
                        onChangeText={setTextValue}
                    /> */}

                    {Object.values(weatherList).length >= 1 && <Text style={styles.subtitle}>Pronostico 5 dias /Hora</Text>}
                    {Object.values(weatherList).length >= 1 && <FlatList
                        horizontal
                        data={weatherList.slice(3)}
                        keyExtractor={(item, index) => index.toString()}
                        renderItem={(data) => {
                            var dt = new Date(data.item.dt * 1000);
                            return <View style={styles.hour}>
                                <Image
                                    style={styles.smallIcon}
                                    source={{
                                        uri: `http://openweathermap.org/img/wn/${data.item.weather[0]?.icon}@4x.png`,
                                    }}
                                />
                                <Text>{Moment(data.item.dt_txt).format('LL')}</Text>
                                <Text>{dt.toLocaleTimeString().replace(/:\d+ /, ' ')}</Text>
                                <Text style={styles.textDetail}>{Math.round(data?.item.main.temp)}°C</Text>
                                <Text>{data.item.weather[0]?.description}</Text>
                            </View>
                        }}
                    />}
                </View>
            </ScrollView>
        </GradientBackground>
    )
}