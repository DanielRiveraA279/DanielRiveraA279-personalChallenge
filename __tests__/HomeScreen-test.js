import React from 'react';
import { shallow } from 'enzyme';
import { render } from 'react-native-testing-library';
import { HomeScreen } from '../src/screens/HomeScreen';

//incorporacion de ejemplo del TDD

describe('HomeScreen', () => {
    let wrapper;

    //setea configuraciones antes de aplicar casos de pruebas
    beforeEach(() => {
        wrapper = shallow(<HomeScreen />);
    });

    it('Render <HomeScreen/>', async () => {

        const renderScreen = render(
            <HomeScreen />
        );
        expect(renderScreen).toBeTruthy();
    })

    it('Testing Components', () => {
        //Verificamos que esten renderizando estas view
        expect(wrapper.find('PickerSearch')).toHaveLength(1); //buscamos los picker que existan
        expect(wrapper.find('Text')).toHaveLength(1);
        expect(wrapper.find('View')).toHaveLength(1);
    });
});