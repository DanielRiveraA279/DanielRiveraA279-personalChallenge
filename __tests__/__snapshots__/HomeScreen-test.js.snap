// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Testing View correctly 1`] = `
<GradientBackground>
  <ScrollView>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 4,
          "marginVertical": 12,
        }
      }
    >
      BUSCAR CLIMA
    </Text>
    <View>
      <PickerSearch
        onValueChange={[Function]}
        select=""
      />
    </View>
  </ScrollView>
</GradientBackground>
`;

exports[`Rendering Testing Component correctly 1`] = `
<GradientBackground>
  <ScrollView>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 4,
          "marginVertical": 12,
        }
      }
    >
      BUSCAR CLIMA
    </Text>
    <View>
      <PickerSearch
        onValueChange={[Function]}
        select=""
      />
    </View>
  </ScrollView>
</GradientBackground>
`;
